source/vir/src/triggers_auto.rs

```diff
@@ -195,23 +200,35 @@ fn trigger_var_depth(ctxt: &Ctxt, term: &Term, depth: u64) -> Option<u64> {
     }
 }
 
-fn make_score(term: &Term, depth: u64) -> Score {
-    Score { depth, size: term_size(term) }
+fn make_score(term: &Term, depth: u64, num_operators: u64) -> Score {
+    Score { depth, size: term_size(term), num_operators }
 }
 
-fn gather_terms(ctxt: &mut Ctxt, ctx: &Ctx, exp: &Exp, depth: u64) -> (bool, Term) {
-    let (is_pure, term) = match &exp.x {
-        ExpX::Const(c) => (true, Arc::new(TermX::App(App::Const(c.clone()), Arc::new(vec![])))),
-        ExpX::Var(x) => (true, Arc::new(TermX::Var(x.clone()))),
+fn gather_terms(ctxt: &mut Ctxt, ctx: &Ctx, exp: &Exp, depth: u64) -> (bool, u64, Term) {
```

Comment:
Rather than computing `num_operators` inside `gather_terms`, I would write a separate function on the `Term` like `term_size`.  Then most of the changes below won't be needed.

Reply:
Oh, you're right. I will make changes.